### PR TITLE
Semi open status

### PR DESCRIPTION
**Pull request by joanvila**
As I mentioned I added the semi-status functionality.

In order to see it live you can clone [this demo repository](https://github.com/joanvila/bigSlide.js-demo).

Here are 2 screenshots demonstrating it

screen shot 2016-04-07 at 17 54 21

screen shot 2016-04-07 at 17 54 10

I also added an option to set the minimum width of the window to let the semi open status work. If the window is thinner, the semi open status will not be applied.

### CODE REVIEW

**Comment by ascott1** [comment]
:heart_eyes_cat:

File: src/jquery.big-slide.js [hunk 1]
```diff
@@ -240,9 +288,22 @@
...
+          this.$shrink.css({
+            'width': 'calc(100% - ' + this.width + ')'
+          });
+          this.$menu.removeClass('semiOpen');
+        } else {
+          this.$menu.css(settings.side, '0');
+          this.$push.css(settings.side, this.width);
+          this.$shrink.css({
+            'width': 'calc(100% - ' + this.width + ')'
+          });
```
Comment (ascott1) [inline src/jquery.big-slide.js]: Could we move this as well as the code to style the closed menu to their own methods? Something like `applyOpenStyles` and `applyCloseStyles`:

For example:

```javascript

applyOpenStyles: function() {
 if (settings.semiOpenStatus && screenWidth > settings.semiOpenScreenWidth) {
    // ...
  } else {
    // ...
  }
}

toggleOpen: function() {
  settings.beforeOpen();	    
  controller.changeState();
  view.applyOpenStyles();
  menuLink.addClass(settings.activeBtn);
  settings.afterOpen();
  // save the state
  if (settings.saveState) {
    localStorage.setItem('bigSlide-savedState', 'open');
  }
}
```

**Comment by joanvila** [comment]
Yes @ascott1 I can do it. 

Actually, I was thinking about the fact of applying the styles with javascript.
What I think that it would be the appropriate, is to have a class for openState, another one for semiOpenState and another one for closedState. And the only thing that we would need to do via javascript would be to toggle those classes. 

The only thing that prevents us from doing this is the fact that the user can choose the menu width with an option and we need to keep changing the width all the time with javascript according to that option.

I'm not sure if the width option is necessary. If not, we could handle all the style changes adding and removing css classes.

**Comment by ascott1** [comment]
I really like that idea @joanvila . I imagine many users could/would set the width via CSS negating the need for the width property.

**Comment by joanvila** [comment]
I've just pushed the `applyOpenStyles` and `applyCloseStyles` methods to isolate the style functions.
The code looks much better now :)

**Comment by joanvila** [comment]
Do you think it can be merged now @ascott1 ?
Once done I will start with what I told about applying the styles via css and not via javascript

**Comment by ascott1** [comment]
Oh, yes! Sorry I've been traveling and lost track of things.

**Comment by joanvila** [comment]
No problem :)

**Pull request merged**